### ISSUE TITLE
Decide if error messages should end with a period

### ISSUE THREAD

**Issue by Minoru** [newsboat/newsboat#879]
In #874, it was found that error messages Newsboat displays rarely (never?) end with periods; sometimes, they end with exclamation marks. It's not clear if there is any rhyme or reason to those choices.

To bring that in order, I think the following has to be done:

1. **try to see a scheme in existing messages**. Perhaps I didn't see it in #874 because I was reviewing other changes. The easiest way to do this, I think, is to look through po/newsboat.pot in search of error messages. Another possibility is to grep for `show_message()`.

2. **find some HCI style guides and see what they recommend**.

**Comment by Minoru** [comment]
@programandala-net left a few remarks about this in https://github.com/newsboat/newsboat/pull/874#issuecomment-612459177:

> In order to check the current situation and decide what to do, consider these remarks:
> 
> If error messages are used as full sentences, they should end with a period. This is even clearer considering that in theory an error message could contain more than one sentence, say:
> 
> ```
> Error: Something is broken. Please restart your system.
> ```
> 
> Currently all of the error messages that don't end with a period are those that end with a paren or a parameter. Therefore a possible explanation is the parameter always includes the period. But an oversight seems more likely.
> 
> In fact there are 3 messages that end with a parameter and a period. See, for example:
> 
> ```
> #: src/controller.cpp:627
> #, c-format
> msgid "An error occurred while parsing %s."
> msgstr "Ocurrió un error al procesar %s."
> ```
> 
> When the ending punctuation is included in the parameter, a note for translators would be helpful, to make sure the punctuation is always coherent.
> 
> See this one, with a paren:
> 
> ```
> #: src/controller.cpp:162 src/controller.cpp:220 src/pbcontroller.cpp:255
> #, c-format
> msgid "Error: an instance of %s is already running (PID: %u)"
> msgstr "Error: una instancia de %s ya está ejecutándose (PID: %u)"
> ```
> 
> Maybe there's a reason in the code for the missing period, I don't know. Anyway, if I recall clearly, in English the period should be before the closing paren. In Spanish it must be always after it.

**Comment by Minoru** [comment]
> If error messages are used as full sentences, they should end with a period.

I'm not convinced of this. There's plenty of cases where full sentences don't end in periods, e.g. titles, slogans on billboards, signs of all sorts. That's why I want to see if there are any human-machine interaction guides out there that specify this.

**Comment by programandala-net** [comment]
> > If error messages are used as full sentences, they should end with a
> > period.
> 
> I'm not convinced of this. There's plenty of cases where full
> sentences don't end in periods, e.g. titles, slogans on billboards,
> signs of all sorts.

Yes, but those are not "messages", which are supposed to be complete
sentences with all their components (even if ommited).

As said, for me the thing is clear, if only because any message could
consist of several sentences:

  Error: something went wrong. Restart the machine.

Why should that message be written without a period?:

  Error: something went wrong. Restart the machine
  Error: something went wrong, restart the machine
  Error: something went wrong; restart the machine

Anyway, as far as I know, both conventions exist about punctuation of
software error messages.

If fact, I think there's a difference between simple, short error
messages used as labels or titles, like HTTP error messages, and
software error messages *explaining* a problem or situation to the user.

> That's why I want to see if there are any human-machine interaction
> guides out there that specify this.

Good idea.

I just happen to find something:

https://style.tidyverse.org/error-messages.html#punctuation

It reads "Errors (...) should end in a full stop", but not all the given
examples follow their own rule :)

-- 
Marcos Cruz
http://programandala.net

**Comment by Minoru** [comment]
One more note I wanted to copy from https://github.com/newsboat/newsboat/pull/874#issuecomment-612459177:

> [Talking of error messages that end with a string parameter, like "An error occurred while parsing %s."] When the ending punctuation is included in the parameter, a note for translators would be helpful, to make sure the punctuation is always coherent.

A note from me: we're [already running xgettext with `-c/`](https://github.com/newsboat/newsboat/blob/7acc6a2e68d4647803e85ef57170aa1fce2e12fd/Makefile#L301), so it's a matter of prefixing such comments with `///` instead of `//`. Looks like xtr (an xgettext-like tool for Rust) extracts any comments, but I didn't test that: https://github.com/woboq/tr/issues/5.
